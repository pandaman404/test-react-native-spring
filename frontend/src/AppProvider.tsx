import { NavigationContainer } from '@react-navigation/native';
import { StatusBar } from 'expo-status-bar';
import { ReactNode } from 'react';
import { SafeAreaProvider } from 'react-native-safe-area-context';

interface AppProviderProps {
  children: ReactNode;
}

export default function AppProvider({ children }: AppProviderProps) {
  return (
    <SafeAreaProvider>
      <NavigationContainer>
        <StatusBar style='auto' />
        {children}
      </NavigationContainer>
    </SafeAreaProvider>
  );
}
