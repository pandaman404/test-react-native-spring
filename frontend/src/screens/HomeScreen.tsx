import { View, StyleSheet } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import Logo from '../components/Logo';
import LinkBtn from '../components/LinkBtn';
import { Colors } from '../constants/colors';

export default function HomeScreen() {
  return (
    <SafeAreaView style={styles.container}>
      <View>
        <Logo />
      </View>
      <View style={styles.linksContainer}>
        <LinkBtn text='Encuesta' pathname='/poll' />
        <LinkBtn text='Resultados' pathname='/results' />
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'center',
    gap: 80,
    paddingVertical: 100,
    backgroundColor: Colors.white,
  },
  linksContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    gap: 50,
    width: '100%',
  },
});
