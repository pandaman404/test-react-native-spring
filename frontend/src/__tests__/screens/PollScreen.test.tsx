import renderer from 'react-test-renderer';
import HomeScreen from '../../screens/HomeScreen';
import AppProvider from '../../AppProvider';

describe('<HomeScreen />', () => {
  it('Should render HomeScreen', () => {
    const tree = renderer
      .create(
        <AppProvider>
          <HomeScreen />
        </AppProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
