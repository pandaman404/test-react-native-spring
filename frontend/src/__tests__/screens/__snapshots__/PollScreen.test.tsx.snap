// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> Should render HomeScreen 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
